great_expectations/util.py

```diff
@@ -1709,14 +1711,12 @@ def convert_ndarray_float_to_datetime_tuple(
     return tuple(convert_ndarray_float_to_datetime_dtype(data=data).tolist())
 
 
-def is_ndarray_decimal_dtype(
-    data: "npt.NDArray",
-) -> TypeGuard["npt.NDArray"]:
+def is_ndarray_decimal_dtype(data: np.ndarray) -> bool:
     """
     Determine whether or not all elements of 1-D "np.ndarray" argument are "decimal.Decimal" type objects.
     """
     value: Any
-    result: bool = all(isinstance(value, decimal.Decimal) for value in data)
+    result: bool = any(isinstance(value, decimal.Decimal) for value in data)
```

Comment:
Why the change from `all` to `any`? The function name `is_ndarray_decimal_dtype` seems to imply that `all` is appropriate since, if we pass this check, we expect all the values to be of the decimal dtype?

Reply:
@billdirks Your example results in having both.  Please see above reply to Gabriel.

Reply:
I don't know if you saw my comment in the thread there. I'd prefer you change this back to `all` and update `isinstance` with a list of allowed values. It seems like 1 `Decimal` doesn't guarantee that the rest are `Decimals` or `floats`.